### ISSUE TITLE
'Access denied' for offer editing

### ISSUE THREAD

**Issue by Vlad-Katsubo** [allegro/allegro-api#5060]
Witam. 
Próbuję wykonać zapytania do API

https://api.allegro.pl/sale/products?ean=8434786979625

https://api.allegro.pl/sale/product-offers/'my-offer's id'.

Otrzymuję taką odpowiedź.

{
    "errors": [
        {
            "code": "AccessDeniedException",
            "message": "Access is denied",
            "details": null,
            "path": null,
            "userMessage": "Access denied."
        }
    ]
}

Używam tokena dla mojego środowiska pracy https://allegro.pl/auth/oauth/token

Czy mógłbyś mi powiedzieć w czym tkwi problem?
![access](https://user-images.githubusercontent.com/92863635/138098518-eb0e72b1-757d-494b-9e13-598e63751a09.png)

**Comment by Lukasz-Zurek** [comment]
Wszystko wskazuje na to, że korzystasz z autoryzacji typu [client-credentials](https://developer.allegro.pl/auth/#clientCredentialsFlow) (bearer-token-for-application), który nie posiada kontekstu użytkownika. Zgodnie z [dokumentacją](https://developer.allegro.pl/documentation/#operation/editProductOffers), musisz być zautoryzowany za pomocą [code](https://developer.allegro.pl/auth/#app) lub [device](https://developer.allegro.pl/auth/#DeviceFlow) flow (bearer-token-for-user).

**Comment by Lukasz-Zurek** [comment]
@Vlad-Katsubo Hello. I am sorry for the long waiting time for a reply.

I have already verified your case. You are using a token which has permissions to a different account which does not have any offers listed. Please decode your token via [this site](https://jwt.io) and pay attention to user_name. It is different than your account where you have the offer which you are trying to edit.

You must be authorized as the seller to whom the offer belongs.

**Comment by Vlad-Katsubo** [comment]
@Lukasz-Zurek  Thank you for the answer. It's Ok.

Okay. It was obvious to me that I'm using a different account's token. I will explain what I want to do again.

I want to create and edit offers and manage my account which is named f******er. As it was said to me in a couple of first messages from your colleague, I can't use -application token for those types of actions. (create, edit). 
I have to get a -user token.
 
So I've created another "application" using my main acc and your tool, got their Client ID and Client Secret and created a link for the user, after redirecting I've received a key in the link, that lasts for 10 seconds and gives opportunity to get another token, that I considered as -user token.

And in my thoughts it was like that:
My main account creates a link using secret credentials, some other account receives permissions to create and edit offers via the token that has been created by key from this link.

Otherwise, why do I even should do all the stuff about Client ID and Client Secret, creating a special link, redirect, get 10 secs lifetime key, creating a token... for what? To just be able to manage a new account? 

Can you please explain to me how can I create, edit and manage my main acc via API?

Which type of verification should I use then.

Thank you.

**Comment by Lukasz-Zurek** [comment]
The most important thing in the authentication & authorization process is the user authorization. We have described it right [here](https://developer.allegro.pl/en/auth/#user-authorization).

Obviously you use a correct flow but please remember that after a correct authentication, the user is redirected to the page where he grants a consent for the application to perform operations/requests on his behalf. You have received a code but you have done it on a different user account. So then after providing this authorization code, you have received an access token for a different user.

I hope that I have described good enough. If not - do not hesitate to ask.

**Comment by Vlad-Katsubo** [comment]
@Lukasz-Zurek 
Am I right to think, that I just have to do the same actions but for the main allegro account - f******er?

**Comment by Lukasz-Zurek** [comment]
@Vlad-Katsubo 

> @Lukasz-Zurek Am I right to think, that I just have to do the same actions but for the main allegro account - f******er?

Yes, exactly. Please read [our guide](https://developer.allegro.pl/en/auth/#user-authorization). Just provide a link in your application which triggers a properly parameterized HTTP request to a resource that allows user authentication.

> Or I can somehow allow this new user to make changes on behalf of my main acc?

No, it is not possible in Allegro API.

**Comment by Vlad-Katsubo** [comment]
@Lukasz-Zurek Thanks for your answer.

Now I'm getting this type of error after clicking on the link with a "properly parameterized HTTP request to a resource that allows user authentication" :(

![image](https://user-images.githubusercontent.com/92863635/140892898-30dffa59-99c3-4acf-9aec-10c5522db5f1.png)

Could you please help?

**Comment by Lukasz-Zurek** [comment]
@Vlad-Katsubo Please make sure that you have provided a correct **redirect_uri**. It could be a problem.

**Comment by Vlad-Katsubo** [comment]
I've got "Client ID" and "Adresy przekierowań" from here https://apps.developer.allegro.pl/. It shouldn't be a problem.

As you maybe remember I've already got a token but for another user. And now I'm trying to do the same but for the right one. The only thing that is different - is that I'm already logged in to Allegro with my main acc. I've also tried to log out and use the authorization link - logging in - and got the same error message with a different "Numer bledu".

What could be wrong? :(

Thank you for your patience.
